src/test/lib/logical_query_plan/lqp_utils_test.cpp

```diff
@@ -305,4 +332,61 @@ TEST_F(LQPUtilsTest, CollectSubqueryExpressionsByLQPNestedSubqueries) {
   EXPECT_EQ(subquery_expressions_by_lqp.find(max_a_subquery->lqp)->second.at(0).lock(), max_a_subquery);
 }
 
+TEST_F(LQPUtilsTest, FindDiamondOriginNode) {
```

Comment:
Maybe add comments to summarize what each test aims to ensure or hint what is the crucial part of the LQP (like in `SimpleDiamondPushdownTest`).